Sources/SwiftProtobuf/ProtobufBinaryTypes.swift

```diff
@@ -25,14 +25,15 @@ public protocol ProtobufBinaryCodableType: ProtobufTypePropertiesBase {
 
     static func setFromProtobufVarint(varint: UInt64, value: inout BaseType?) throws -> Bool
     static func setFromProtobufVarint(varint: UInt64, value: inout [BaseType]) throws -> Bool
```

Comment:
Look at `ProtobufFieldWireTypeVarint` in ProtobufBinaryDecoding.swift to see how these specific returns are used, and `extension ProtobufEnum` in ProtobufBinaryTypes.swift to see the only place that actually sets them.

The other specific variation is in `setFromProtobufBuffer` which passes in a `Data` object to receive any leftovers.

The variation among the `setFrom` methods here is not ideal, but I've not come up with an approach I like better.  This does fix the unknown enum handling and it does isolate this issue, so I think it's an improvement.  If you see a way to make it cleaner, please let me know.